CONTRIBUTING.MD

```diff
@@ -0,0 +1,186 @@
+
+Some examples of valid commit messages (used as first lines):
+
+> - feat(account-client): add method XXX
+> - chore(deps): update dependency XXX to v3.0.7
+> - fix(search_user_ids): rename parameter clusterName
+> - chore: reword contributions guides
+
+## Branch organization
+
+The project is based on the classic GitHub flow:
+
+- `master` for the current version being worked on – Pull requests for bugs and feature related to the current major version should be created against this branch
+- `vX` for each major version (`X` being a number) – Pull requests for critical bug fixes should be created against this branch
+
+Most of the time, your pull requests should target the `master` branch.
+
+_Note that no new features will be developed or backported for the `vX` branches._
+
+## Requirements
+
+To run this project, you will need:
+
+- Ruby ≥ 2.2
+- [Bundler](https://bundler.io/)
+
+## Use the Dockerfile
+
+If you want to contribute to this project without installing all its dependencies, you can use our Docker image. 
+Please check our [dedicated guide](DOCKER_README.MD) to learn more.
+
+## Code structure
+
+Here are the main files and folders of the project
+
+```
+▸ lib/algolia/              << standalone clients and helpers classes
+▸ lib/test/                 << gathers the unit and integration tests
+  .rubocop.yml              << contains the rule used for the linter
+  algolia.gemspec           << gemspec file
+  CHANGELOG.md              << changelog file
+  CONTRIBUTING.md           << this file
+  Gemfile                   << dependencies needed to run the project locally
+  Rakefile                  << defines the different tasks to lint/test the project
+  README.md                 << the introduction of the project
+```
+
+### The lib/algolia folder
+
+```
+▸ lib/algolia/
+    ▸ config/       << the configurations associated with each clients 
+    ▸ enums/        << enumerables used accross the lib
+    ▸ http/         << the http layer logic
+    ▸ iterators/    << the iterators used for the browsing methods
+    ▸ responses/    << the waitable responses
+    ▸ transport/    << the transport layer logic
+```
+
+## Tests
+
+Our unit and integration tests are written with [Minitest](https://github.com/seattlerb/minitest), in the `test` syntax.
+
+To run all the tests at once:
+
+```sh
+bundle exec rake test:all
+```
+
+To run only the unit tests:
+
+```sh
+bundle exec rake test:unit
+```
+
+To run only the integration tests:
+
+```sh
+bundle exec rake test:integration
+```
+
+We ask that for each fix or feature submitted, you add at least one test demonstrating its behaviour. As you will need to use your own Algolia
+credentials to run them, we advise you to keep them short and use a small dataset, as well as using a mock requester anytime it's possible. If you 
+encounter huge data overload because of testing, please reach out to [our support team](support@algolia.com).
```

Comment:
Can you please add a link on the `using a mock requester` to link to the docs on how to do that?
```suggestion
We ask that for each fix or feature submitted, you add at least one test demonstrating its behaviour. As you will need to use your own Algolia credentials to run them, we advise you to keep them short and use a small dataset, as well as using a mock requester anytime it's possible. If you encounter huge data overload because of testing, please reach out to [our support team](support@algolia.com).
```